### PR TITLE
Removal of EOL deprecations for 23.05

### PR DESCRIPTION
**Pull request by rustyrussell**
This is our first set of removals under the new deprecation system: I've only removed things which were officially end-of-lifed in 2023.  None of these are present in the current release unless someone manually enabled with `i-promise-to-fix-broken-api-user`, so hopefully they will have complained before this release if we decide to delay/reinstate APIs.

### CODE REVIEW

**Comment by rustyrussell** [comment]
Rebased on master with doc reorg!

File: connectd/connectd.c [hunk 1]
```diff
@@ -1272,42 +1272,6 @@ setup_listeners(const tal_t *ctx,
 		return NULL;
 	}
 
-	/* If we want websockets to match IPv4/v6, set it up now. */
-	if (daemon->websocket_port) {
```
Comment (vincenzopalazzo) [inline connectd/connectd.c:1276]: looks like this experiment of yours is redacted also from the BOLTs?

File: plugins/commando.c [hunk 2]
```diff
@@ -397,8 +397,7 @@ static void try_command(struct commando *incoming STEALS)
 	}
 	filter = json_get_member(buf, toks, "filter");
 	id = json_get_member(buf, toks, "id");
-	if (!id && !command_deprecated_in_nocmd_ok(plugin, "commando.missing-id",
-						   "v23.02", "v24.02")) {
+	if (!id) {
```
Comment (vincenzopalazzo) [inline plugins/commando.c:400]: uh! I should update my proxy for next time :) thanks for reminding me

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Concept-ACK 13af9547c11b8d3c181a7c91832039f3c16ab2ea

Thanks for this deprecation series, looks amazing, but something (assuming in listpeers) is dirty, the CI found the following one

```
=============================== warnings summary ===============================
../../../../../../../opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/cheroot/__init__.py:7
  /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/cheroot/__init__.py:7: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

../../../../../../../opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pkg_resources/__init__.py:2871
../../../../../../../opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pkg_resources/__init__.py:2871
  /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('zope')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================== 1 passed, 3 warnings in 6.60s =========================
make[1]: Leaving directory '/home/runner/work/lightning/lightning/contrib/pyln-testing'
lightningd/peer_control.c:2321:9: error: Uninitialized variable: channel [uninitvar]
   if (!channel->scb)
        ^
lightningd/peer_control.c:3117:39: error: Uninitialized variable: channel [uninitvar]
    if (!channel_state_can_setchannel(channel->state))
                                      ^
lightningd/peer_control.c:1389:11: error: Uninitialized variable: channel [uninitvar]
          channel->error)));
          ^
make: *** [Makefile:548: check-cppcheck] Error 123
rm external/build-x86_64-linux-gnu/libwally-core-build/src/secp256k1/libsecp256k1.la
```

File: connectd/connectd.c (hunk 1 above)
Comment (rustyrussell) [reply connectd/connectd.c:1276]: It was: I withdrew the proposal.

**Comment by rustyrussell** [comment]
> Concept-ACK [13af954](https://github.com/ElementsProject/lightning/commit/13af9547c11b8d3c181a7c91832039f3c16ab2ea)
> 
> Thanks for this deprecation series, looks amazing, but something (assuming in listpeers) is dirty, the CI found the following one
> 
> ```
> =============================== warnings summary ===============================
> ../../../../../../../opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/cheroot/__init__.py:7
>   /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/cheroot/__init__.py:7: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
>     import pkg_resources
> 
> ../../../../../../../opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pkg_resources/__init__.py:2871
> ../../../../../../../opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pkg_resources/__init__.py:2871
>   /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('zope')`.
>   Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
>     declare_namespace(pkg)
> 
> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> ======================== 1 passed, 3 warnings in 6.60s =========================
> make[1]: Leaving directory '/home/runner/work/lightning/lightning/contrib/pyln-testing'
> lightningd/peer_control.c:2321:9: error: Uninitialized variable: channel [uninitvar]
>    if (!channel->scb)
>         ^
> lightningd/peer_control.c:3117:39: error: Uninitialized variable: channel [uninitvar]
>     if (!channel_state_can_setchannel(channel->state))
>                                       ^
> lightningd/peer_control.c:1389:11: error: Uninitialized variable: channel [uninitvar]
>           channel->error)));
>           ^
> make: *** [Makefile:548: check-cppcheck] Error 123
> rm external/build-x86_64-linux-gnu/libwally-core-build/src/secp256k1/libsecp256k1.la
> ```

Yes, cppcheck is broken.  It cannot handle our macros, so we try to suppress some and it's not reliable.  It's been quite a while since it has actually found any bugs, so I will remove it.

**Comment by rustyrussell** [comment]
> > lightningd/peer_control.c:2321:9: error: Uninitialized variable: channel [uninitvar]
> >    if (!channel->scb)
> >         ^
> > lightningd/peer_control.c:3117:39: error: Uninitialized variable: channel [uninitvar]
> >     if (!channel_state_can_setchannel(channel->state))
> >                                       ^
> > lightningd/peer_control.c:1389:11: error: Uninitialized variable: channel [uninitvar]
> >           channel->error)));
> >           ^
> > make: *** [Makefile:548: check-cppcheck] Error 123
> > rm external/build-x86_64-linux-gnu/libwally-core-build/src/secp256k1/libsecp256k1.la
> > ```
> 
> Yes, cppcheck is broken. It cannot handle our macros, so we try to suppress some and it's not reliable. It's been quite a while since it has actually found any bugs, so I will remove it.

OK, I'll try adding suppressions first...

**Pull request merged**